webapp/src/shared/hooks/UseIsElementVisible.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+
+export function UseIsElementVisible() {
+  const [isVisible, setIsVisible] = React.useState(false);
+  const observer = React.useRef<IntersectionObserver | null>();
```

Comment:
praise: Now we can reuse it in more places, good job :+1: 

suggestion: I think the convention for custom hooks is to start with lowercase letters -> `useIsElementVisible`?

Reply:
done!